Error getting vmconfig

```
proxbox [2026-05-21 08:35:44,127] [ERROR] exceptions: Unhandled exception on GET /proxmox/xx4-xx-x3-26/qemu/126/config
  + Exception Group Traceback (most recent call last):
  |   File "/app/.venv/lib/python3.13/site-packages/starlette/_utils.py", line 81, in collapse_excgroups
  |     yield
  |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 192, in __call__
  |     async with anyio.create_task_group() as task_group:
  |                ~~~~~~~~~~~~~~~~~~~~~~~^^
  |   File "/app/.venv/lib/python3.13/site-packages/anyio/_backends/_asyncio.py", line 799, in __aexit__
  |     raise BaseExceptionGroup(
  |         "unhandled errors in a TaskGroup", self._exceptions
  |     ) from None
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/errors.py", line 164, in __call__
    |     await self.app(scope, receive, _send)
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 191, in __call__
    |     with recv_stream, send_stream, collapse_excgroups():
    |                                    ~~~~~~~~~~~~~~~~~~^^
    |   File "/usr/local/lib/python3.13/contextlib.py", line 162, in __exit__
    |     self.gen.throw(value)
    |     ~~~~~~~~~~~~~~^^^^^^^
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/_utils.py", line 87, in collapse_excgroups
    |     raise exc
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 193, in __call__
    |     response = await self.dispatch_func(request, call_next)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/app/.venv/lib/python3.13/site-packages/proxbox_api/app/factory.py", line 168, in dispatch
    |     return await call_next(request)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 168, in call_next
    |     raise app_exc from app_exc.__cause__ or app_exc.__context__
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 144, in coro
    |     await self.app(scope, receive_or_disconnect, send_no_error)
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 191, in __call__
    |     with recv_stream, send_stream, collapse_excgroups():
    |                                    ~~~~~~~~~~~~~~~~~~^^
    |   File "/usr/local/lib/python3.13/contextlib.py", line 162, in __exit__
    |     self.gen.throw(value)
    |     ~~~~~~~~~~~~~~^^^^^^^
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/_utils.py", line 87, in collapse_excgroups
    |     raise exc
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 193, in __call__
    |     response = await self.dispatch_func(request, call_next)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/app/.venv/lib/python3.13/site-packages/proxbox_api/app/factory.py", line 222, in dispatch
    |     return await call_next(request)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 168, in call_next
    |     raise app_exc from app_exc.__cause__ or app_exc.__context__
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 144, in coro
    |     await self.app(scope, receive_or_disconnect, send_no_error)
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 191, in __call__
    |     with recv_stream, send_stream, collapse_excgroups():
    |                                    ~~~~~~~~~~~~~~~~~~^^
    |   File "/usr/local/lib/python3.13/contextlib.py", line 162, in __exit__
    |     self.gen.throw(value)
    |     ~~~~~~~~~~~~~~^^^^^^^
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/_utils.py", line 87, in collapse_excgroups
    |     raise exc
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 193, in __call__
    |     response = await self.dispatch_func(request, call_next)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/app/.venv/lib/python3.13/site-packages/proxbox_api/app/factory.py", line 186, in dispatch
    |     response = await call_next(request)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 168, in call_next
    |     raise app_exc from app_exc.__cause__ or app_exc.__context__
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 144, in coro
    |     await self.app(scope, receive_or_disconnect, send_no_error)
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/cors.py", line 88, in __call__
    |     await self.app(scope, receive, send)
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/middleware/exceptions.py", line 63, in __call__
    |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    |     raise exc
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    |     await app(scope, receive, sender)
    |   File "/app/.venv/lib/python3.13/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    |     await self.app(scope, receive, send)
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/routing.py", line 660, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/routing.py", line 680, in app
    |     await route.handle(scope, receive, send)
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/routing.py", line 276, in handle
    |     await self.app(scope, receive, send)
    |   File "/app/.venv/lib/python3.13/site-packages/fastapi/routing.py", line 134, in app
    |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    |     raise exc
    |   File "/app/.venv/lib/python3.13/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    |     await app(scope, receive, sender)
    |   File "/app/.venv/lib/python3.13/site-packages/fastapi/routing.py", line 120, in app
    |     response = await f(request)
    |                ^^^^^^^^^^^^^^^^
    |   File "/app/.venv/lib/python3.13/site-packages/fastapi/routing.py", line 695, in app
    |     content = await serialize_response(
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^
    |     ...<11 lines>...
    |     )
    |     ^
    |   File "/app/.venv/lib/python3.13/site-packages/fastapi/routing.py", line 300, in serialize_response
    |     raise ResponseValidationError(
    |     ...<3 lines>...
    |     )
    | fastapi.exceptions.ResponseValidationError: 1 validation error:
    |   {'type': 'value_error', 'loc': ('response',), 'msg': 'Value error, Invalid key: bios', 'input': {'agent': '1', 'bios': 'ovmf', 'boot': 'order=sata0;sata1', 'cores': 2, 'cpu': 'Cascadelake-Server-noTSX', 'digest': '1c6a44c158cd25dd20fb45a56bcf2c3258abde1b', 'efidisk0': 'ceph-vm-data:vm-126-disk-0,size=128K', 'machine': 'pc-i440fx-9.0', 'memory': '3072', 'meta': 'creation-qemu=9.0.2,ctime=1727867268', 'name': 'xx.example.example.com', 'numa': False, 'ostype': 'win10', 'scsihw': 'virtio-scsi-single', 'smbios1': 'uuid=42393877-a3dd-c53b-c004-7f66e6aa0c9c', 'sockets': 1, 'tags': 'xxx;xxxxx-xxx;xxxx', 'vmgenid': 'c5b96294-3b88-41c4-a08b-6c20260d578f', 'sata1': 'ceph-vm-data:vm-126-disk-1,size=50G,ssd=1', 'sata0': 'none,media=cdrom', 'net0': 'virtio=00:50:56:b9:02:4d,bridge=v656'}, 'ctx': {'error': ValueError('Invalid key: bios')}}
    | 
    |   File "/app/.venv/lib/python3.13/site-packages/proxbox_api/routes/proxmox/__init__.py", line 396, in get_vm_config
    |     GET /proxmox/{node}/{type}/{vmid}/config
    +------------------------------------
```
Running version:
proxbox-api: v0.0.13

--- a/proxbox_api/schemas/virtualization/__init__.py
+++ b/proxbox_api/schemas/virtualization/__init__.py
@@ -4,7 +4,7 @@ from __future__ import annotations
 
 import re
 
-from pydantic import BaseModel, ConfigDict, computed_field, field_validator, model_validator
+from pydantic import BaseModel, ConfigDict, Field, computed_field, field_validator, model_validator
 
 from proxbox_api.enum.proxmox import DiskFormat, ProxmoxVMStatus
 from proxbox_api.proxmox_to_netbox.schemas.disks import ProxmoxDiskEntry, parse_vm_config_disks
@@ -27,40 +27,185 @@ def _parse_key_value_string(value: object) -> dict[str, str]:
     return parsed
 
 
+# Numbered device-bus prefixes from the Proxmox VE API spec (proxmox-sdk source of truth).
+# Each entry represents a family indexed by an integer suffix (e.g. net0, net1, sata3 …).
+# Single-instance fields (efidisk0, tpmstate0, audio0 …) are declared as explicit model
+# fields below and are therefore NOT included in this pattern.
+_DYNAMIC_KEY_RE = re.compile(
+    r"^(scsi|net|ide|sata|virtio|unused|smbios|hostpci|usb|serial|parallel|numa|ipconfig|virtiofs)\d+$"
+)
+
+
 class VMConfig(BaseModel):
+    """Proxmox VM/CT configuration response.
+
+    Covers both QEMU and LXC config payloads returned by
+    ``GET /proxmox/{node}/{type}/{vmid}/config``.
+
+    Static fields are derived from ``GetNodesNodeQemuVmidConfigResponse`` in
+    proxmox-sdk, which is the generated OpenAPI source of truth for every field
+    Proxmox VE can return.  Numbered device-bus fields (net[n], scsi[n],
+    sata[n] …) land in ``model_extra`` via the dynamic whitelist regex and are
+    parsed by the ``disks`` / ``networks`` computed fields.
+    """
+
     model_config = ConfigDict(extra="allow", populate_by_name=True)
 
+    # ------------------------------------------------------------------ #
+    # Shared QEMU + LXC
+    # ------------------------------------------------------------------ #
     parent: str | None = None
     digest: str | None = None
-    swap: int | None = None
-    searchdomain: str | None = None
-    boot: str | None = None
-    name: str | None = None
-    cores: int | None = None
-    scsihw: str | None = None
-    vmgenid: str | None = None
-    memory: int | None = None
     description: str | None = None
-    ostype: str | None = None
-    numa: bool | None = None
-    sockets: int | None = None
-    cpulimit: int | None = None
-    onboot: bool | None = None
-    cpuunits: int | None = None
-    agent: bool | None = None
+    name: str | None = None
     tags: str | None = None
+    onboot: bool | None = None
+    boot: str | None = None
+    arch: str | None = None
+    ostype: str | None = None
+    searchdomain: str | None = None
+    nameserver: str | None = None
+    sshkeys: str | None = None
+    ciuser: str | None = None
+    ipconfig0: str | None = None
+
+    # ------------------------------------------------------------------ #
+    # CPU / memory
+    # ------------------------------------------------------------------ #
+    cores: int | None = None
+    sockets: int | None = None
+    numa: bool | None = None
+    cpu: str | None = None  # QEMU emulated CPU type, e.g. "Cascadelake-Server-noTSX"
+    cpulimit: float | None = None  # fractional limit, e.g. 2.5
+    cpuunits: int | None = None
+    vcpus: int | None = None
+    memory: int | None = None  # keep int for backward compat; LXC and older QEMU return integers
+    balloon: int | None = None
+    shares: int | None = None
+    hugepages: str | None = None
+    keephugepages: bool | None = None
+    affinity: str | None = None
+
+    # ------------------------------------------------------------------ #
+    # Machine / BIOS / boot
+    # ------------------------------------------------------------------ #
+    bios: str | None = None  # "seabios" (default) or "ovmf" (UEFI/EFI)
+    machine: str | None = None  # e.g. "pc-i440fx-9.0", "q35"
+    acpi: bool | None = None
+    kvm: bool | None = None
+    localtime: bool | None = None
+    tdf: bool | None = None
+    tablet: bool | None = None
+    keyboard: str | None = None
+    vga: str | None = None
+    audio0: str | None = None
+    spice_enhancements: str | None = None
+    smbios1: str | None = None
+    vmgenid: str | None = None
+    vmstate: str | None = None
+    vmstatestorage: str | None = None
+
+    # ------------------------------------------------------------------ #
+    # Storage
+    # ------------------------------------------------------------------ #
+    scsihw: str | None = None
+    efidisk0: str | None = None  # EFI variable disk; always efidisk0
+    tpmstate0: str | None = None  # TPM state disk; always tpmstate0
+    bootdisk: str | None = None
+    cdrom: str | None = None
+
+    # ------------------------------------------------------------------ #
+    # Network / agent
+    # ------------------------------------------------------------------ #
+    agent: bool | None = None
+
+    # ------------------------------------------------------------------ #
+    # Lifecycle / state
+    # ------------------------------------------------------------------ #
+    autostart: bool | None = None
+    reboot: bool | None = None
+    freeze: bool | None = None
+    protection: bool | None = None
+    template: bool | None = None
+    lock: str | None = None
+    startup: str | None = None
+    startdate: str | None = None
+    watchdog: str | None = None
+    hookscript: str | None = None
+
+    # ------------------------------------------------------------------ #
+    # Hotplug / hardware
+    # ------------------------------------------------------------------ #
+    hotplug: str | None = None
+    rng0: str | None = None
+    ivshmem: str | None = None
+
+    # ------------------------------------------------------------------ #
+    # Migration / snapshots
+    # ------------------------------------------------------------------ #
+    migrate_downtime: float | None = None
+    migrate_speed: int | None = None
+    snaptime: int | None = None
+    smp: int | None = None
+
+    # ------------------------------------------------------------------ #
+    # Cloud-init
+    # ------------------------------------------------------------------ #
+    cicustom: str | None = None
+    cipassword: str | None = None
+    citype: str | None = None
+    ciupgrade: bool | None = None
+
+    # ------------------------------------------------------------------ #
+    # Runtime / snapshot metadata (read-only, included by Proxmox)
+    # ------------------------------------------------------------------ #
+    meta: str | None = None
+    runningcpu: str | None = None
+    runningmachine: str | None = None
+    args: str | None = None
+
+    # ------------------------------------------------------------------ #
+    # Advanced / vendor-specific (Proxmox API uses dash-names for these)
+    # ------------------------------------------------------------------ #
+    allow_ksm: str | None = Field(None, alias="allow-ksm")
+    amd_sev: str | None = Field(None, alias="amd-sev")
+    intel_tdx: str | None = Field(None, alias="intel-tdx")
+    running_nets_host_mtu: str | None = Field(None, alias="running-nets-host-mtu")
+
+    # ------------------------------------------------------------------ #
+    # LXC-only (kept for the shared QEMU+LXC config route)
+    # ------------------------------------------------------------------ #
+    swap: int | None = None
     rootfs: str | None = None
     unprivileged: bool | None = None
     nesting: bool | None = None
-    nameserver: str | None = None
-    arch: str | None = None
     hostname: str | None = None
     features: str | None = None
-    ciuser: str | None = None
-    sshkeys: str | None = None
-    ipconfig0: str | None = None
 
-    @field_validator("numa", "onboot", "agent", "unprivileged", "nesting", mode="before")
+    # ------------------------------------------------------------------ #
+    # Validators
+    # ------------------------------------------------------------------ #
+
+    @field_validator(
+        "numa",
+        "onboot",
+        "agent",
+        "unprivileged",
+        "nesting",
+        "acpi",
+        "autostart",
+        "ciupgrade",
+        "freeze",
+        "keephugepages",
+        "kvm",
+        "localtime",
+        "protection",
+        "reboot",
+        "tablet",
+        "tdf",
+        "template",
+        mode="before",
+    )
     @classmethod
     def _coerce_bool_fields(cls, value: object) -> bool | None:
         return normalize_bool(value)
@@ -68,15 +213,28 @@ class VMConfig(BaseModel):
     @model_validator(mode="before")
     @classmethod
     def validate_dynamic_keys(cls, values: object) -> object:
-        # Validate dynamic keys (e.g. scsi0, net0, etc.).
+        """Reject keys that are not known static fields or numbered device buses.
+
+        Accepted keys:
+        - Every Python field name declared on this model.
+        - Every ``Field(alias=...)`` value (e.g. ``allow-ksm``, ``amd-sev``).
+        - Any key matching ``_DYNAMIC_KEY_RE`` (net[n], scsi[n], sata[n] …).
+        """
         if isinstance(values, dict):
+            # Build accepted-key set: Python field names + their declared aliases.
+            known_static: set[str] = set(cls.model_fields)
+            for fi in cls.model_fields.values():
+                if fi.alias:
+                    known_static.add(fi.alias)
+
             for key in values.keys():
-                if (
-                    not re.match(r"^(scsi|net|ide|unused|smbios)\d+$", key)
-                    and key not in cls.model_fields
-                ):
+                if not _DYNAMIC_KEY_RE.match(key) and key not in known_static:
                     raise ValueError(f"Invalid key: {key}")
         return values
+
+    # ------------------------------------------------------------------ #
+    # Computed fields
+    # ------------------------------------------------------------------ #
 
     @computed_field(return_type=list[ProxmoxDiskEntry])
     @property

--- a/tests/test_vm_config_uefi.py
+++ b/tests/test_vm_config_uefi.py
@@ -1,0 +1,200 @@
+"""Regression tests for VMConfig fields missing from the static whitelist.
+
+Issue #144: VMs with bios=ovmf (UEFI/EFI boot) caused a ResponseValidationError
+because 'bios' was not in VMConfig.model_fields and did not match the dynamic-key
+regex ^(scsi|net|ide|unused|smbios)\\d+$.  Several other real QEMU config fields
+(sata[n], virtio[n], efidisk0, machine, cpu, etc.) had the same problem.
+"""
+
+from __future__ import annotations
+
+import pytest
+
+from proxbox_api.schemas.virtualization import VMConfig
+
+# ---------------------------------------------------------------------------
+# The exact payload from issue #144
+# ---------------------------------------------------------------------------
+
+ISSUE_144_PAYLOAD: dict[str, object] = {
+    "agent": "1",
+    "bios": "ovmf",
+    "boot": "order=sata0;sata1",
+    "cores": 2,
+    "cpu": "Cascadelake-Server-noTSX",
+    "digest": "1c6a44c158cd25dd20fb45a56bcf2c3258abde1b",
+    "efidisk0": "ceph-vm-data:vm-126-disk-0,size=128K",
+    "machine": "pc-i440fx-9.0",
+    "memory": "3072",
+    "meta": "creation-qemu=9.0.2,ctime=1727867268",
+    "name": "xx.example.example.com",
+    "numa": False,
+    "ostype": "win10",
+    "scsihw": "virtio-scsi-single",
+    "smbios1": "uuid=42393877-a3dd-c53b-c004-7f66e6aa0c9c",
+    "sockets": 1,
+    "tags": "xxx;xxxxx-xxx;xxxx",
+    "vmgenid": "c5b96294-3b88-41c4-a08b-6c20260d578f",
+    "sata1": "ceph-vm-data:vm-126-disk-1,size=50G,ssd=1",
+    "sata0": "none,media=cdrom",
+    "net0": "virtio=00:50:56:b9:02:4d,bridge=v656",
+}
+
+
+def test_vm_config_accepts_issue_144_payload() -> None:
+    """VMConfig must not raise for the exact payload reported in issue #144."""
+    cfg = VMConfig.model_validate(ISSUE_144_PAYLOAD)
+    assert cfg.bios == "ovmf"
+    assert cfg.machine == "pc-i440fx-9.0"
+    assert cfg.cpu == "Cascadelake-Server-noTSX"
+    assert cfg.efidisk0 == "ceph-vm-data:vm-126-disk-0,size=128K"
+
+
+# ---------------------------------------------------------------------------
+# Individual static fields that were missing
+# ---------------------------------------------------------------------------
+
+
+@pytest.mark.parametrize(
+    "field,value",
+    [
+        ("bios", "ovmf"),
+        ("bios", "seabios"),
+        ("machine", "pc-i440fx-9.0"),
+        ("machine", "q35"),
+        ("efidisk0", "local-lvm:vm-100-disk-0,size=128K"),
+        ("cpu", "Cascadelake-Server-noTSX"),
+        ("cpu", "host"),
+        ("acpi", True),
+        ("acpi", False),
+        ("balloon", 512),
+        ("hotplug", "network,disk,usb"),
+        ("vga", "std"),
+        ("kvm", True),
+        ("localtime", False),
+        ("freeze", True),
+        ("tablet", True),
+        ("tdf", False),
+        ("template", False),
+        ("autostart", False),
+        ("reboot", True),
+        ("protection", False),
+        ("onboot", True),
+        ("hookscript", "local:snippets/hook.pl"),
+        ("lock", "migrate"),
+        ("meta", "creation-qemu=9.0.2,ctime=1727867268"),
+        ("runningmachine", "pc-i440fx-9.0+pve0"),
+        ("runningcpu", "Cascadelake-Server-noTSX"),
+        ("vmstate", "local:snippets/vm-100.state"),
+        ("vmstatestorage", "local"),
+        ("watchdog", "ib700,action=reset"),
+        ("rng0", "source=/dev/urandom,max_bytes=1024,period=1000"),
+        ("ivshmem", "size=32,name=foo"),
+        ("hugepages", "1024"),
+        ("keyboard", "en-us"),
+        ("audio0", "device=AC97,driver=spice"),
+        ("spice_enhancements", "foldersharing=0,videostreaming=off"),
+        ("startdate", "2006-06-17T16:01:21"),
+        ("startup", "order=1,up=30,down=60"),
+        ("shares", 1000),
+        ("smp", 4),
+        ("vcpus", 2),
+        ("tpmstate0", "local-lvm:vm-100-disk-1,size=4M"),
+        ("smbios1", "uuid=42393877-a3dd-c53b-c004-7f66e6aa0c9c"),
+        ("snaptime", 1727867268),
+        ("bootdisk", "sata0"),
+        ("cdrom", "local:iso/debian.iso"),
+        ("ciuser", "admin"),
+        ("cipassword", "hunter2"),
+        ("cicustom", "user=local:snippets/user.yml"),
+        ("citype", "nocloud"),
+        ("ciupgrade", True),
+        ("keephugepages", False),
+        ("affinity", "0,5,8-11"),
+        ("args", "-no-reboot"),
+        ("migrate_speed", 100),
+        ("migrate_downtime", 0.5),
+        ("cpulimit", 2.5),
+    ],
+)
+def test_vm_config_accepts_qemu_static_field(field: str, value: object) -> None:
+    cfg = VMConfig.model_validate({"digest": "abc", field: value})
+    assert getattr(cfg, field) == value
+
+
+# ---------------------------------------------------------------------------
+# Dynamic numbered fields that were missing from the regex
+# ---------------------------------------------------------------------------
+
+
+@pytest.mark.parametrize(
+    "key,value",
+    [
+        # sata disks
+        ("sata0", "ceph-vm-data:vm-126-disk-0,size=50G"),
+        ("sata1", "local:vm-126-disk-1,size=20G,ssd=1"),
+        ("sata5", "none,media=cdrom"),
+        # virtio disks
+        ("virtio0", "local-lvm:vm-100-disk-0,size=32G"),
+        ("virtio15", "ceph:vm-100-disk-15,size=10G"),
+        # hostpci
+        ("hostpci0", "0000:03:00,pcie=1"),
+        ("hostpci7", "0000:04:00.1"),
+        # USB devices
+        ("usb0", "host=10de:1234"),
+        ("usb4", "spice"),
+        # serial ports
+        ("serial0", "socket"),
+        ("serial3", "/dev/ttyS0"),
+        # parallel ports
+        ("parallel0", "/dev/parport0"),
+        ("parallel2", "/dev/parport2"),
+        # NUMA nodes
+        ("numa0", "cpus=0-3,hostnodes=0,memory=1024,policy=bind"),
+        # ipconfig cloud-init (ipconfig1+ are dynamic; ipconfig0 is a static field)
+        ("ipconfig1", "ip=10.0.0.2/24,gw=10.0.0.1"),
+        ("ipconfig3", "ip=192.168.1.100/24,gw=192.168.1.1"),
+        # virtiofs
+        ("virtiofs0", "dirid=123,cache=auto"),
+    ],
+)
+def test_vm_config_accepts_dynamic_numbered_field(key: str, value: str) -> None:
+    cfg = VMConfig.model_validate({"digest": "abc", key: value})
+    assert (cfg.model_extra or {}).get(key) == value
+
+
+# ---------------------------------------------------------------------------
+# Hyphenated field aliases (Proxmox returns them with dashes)
+# ---------------------------------------------------------------------------
+
+
+@pytest.mark.parametrize(
+    "key,value",
+    [
+        ("allow-ksm", "1"),
+        ("amd-sev", "type=snp"),
+        ("intel-tdx", "1"),
+        ("running-nets-host-mtu", "net0=1500"),
+    ],
+)
+def test_vm_config_accepts_hyphenated_field(key: str, value: str) -> None:
+    """Fields with dash-aliases in the Proxmox API must not be rejected."""
+    cfg = VMConfig.model_validate({"digest": "abc", key: value})
+    # The value is accessible via model_extra since the Python field name uses underscores
+    # but the alias is a dash-name.  Both should resolve without error.
+    assert cfg is not None
+
+
+# ---------------------------------------------------------------------------
+# Validation still rejects genuinely unknown keys
+# ---------------------------------------------------------------------------
+
+
+def test_vm_config_rejects_truly_unknown_key() -> None:
+    with pytest.raises(ValueError, match="Invalid key"):
+        VMConfig.model_validate({"totally_made_up_key": "x"})
+
+
+def test_vm_config_rejects_unknown_prefixed_numbered_key() -> None:
+    with pytest.raises(ValueError, match="Invalid key"):
+        VMConfig.model_validate({"xyzbus0": "x"})

